apps/foundation-demo/src/app/checkout/components/shipping-async-wrapper/shipping-async-wrapper.component.ts

```diff
@@ -0,0 +1,19 @@
+import { Component, Input } from '@angular/core';
+
+@Component({
+  selector: 'shipping-async-wrapper',
+  templateUrl: './shipping-async-wrapper.component.html'
+})
+export class ShippingAsyncWrapperComponent {
+
+  @Input() isShippingInfoValid: Boolean;
+  showShippingForm: boolean;
+
+  ngOnInit() {
+    this.showShippingForm = !this.isShippingInfoValid;
+  }
+
+  toggleShippingView() {
```

Comment:
This needs to be stored in redux state that is appended by foundation-demo.

Reply:
done